pkg/jetstream/subscription.go

```diff
@@ -0,0 +1,44 @@
+package jetstream
+
+import (
+	"context"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+// subscription provides a self-managing wrapper for an individual jetstream subscription
+type subscription struct {
+	cc      jetstream.ConsumeContext
+	output  chan *message.Message
+	closing chan struct{}
+}
+
+func newSubscription(closing chan struct{}, output chan *message.Message, options ...func(*subscription)) *subscription {
```

Comment:
Pass in closing func here instead of with context to listen.  Or make listen a pure function (subscribe)